### DISCUSSION TITLE
Problem with setting values in store when using Composition API setup.

### DISCUSSION THREAD

**Discussion by Miksser** [vuejs/pinia#2057] [Help and Questions] [answered]
### Reproduction

https://stackblitz.com/edit/nuxt-starter-efm7cs?file=app.vue

### Steps to reproduce the bug

- Use <script setup lang="ts"> to initialize the store with the code: const countStore = useCountStore();
- Increment the count variable with count++.

### Expected behavior

- The value will be increased by 1 unit. The application will display 1.

### Actual behavior

- The value is increased by 2 units.

### Additional information

**package.json**
```
{
  "devDependencies": {
    "nuxt": "^3.2.3"
  },
  "dependencies": {
    "@pinia/nuxt": "^0.4.7"
  }
}
```

**nuxt.config.ts**
```
export default defineNuxtConfig({
  modules: ['@pinia/nuxt'],
  pinia: {
    autoImports: [['defineStore', 'definePiniaStore']],
  },
  imports: {
    dirs: ['stores'],
  },
});
```
**stores/CountStore.js**
```
export const useCountStore = definePiniaStore('CountStore', {
  state: () => {
    return {
      count: 0,
    };
  }
});
```

**app.vue**
```
<script setup lang="ts">
const countStore = useCountStore();
countStore.count++;
</script>
<template>
  <div>
    {{ countStore.count }}
  </div>
</template>
```

**Comment by posva** [comment, marked as answer]
It's increment on the server and then on the clientt. That's why you see 2.

**Reply by Miksser** [reply]
Thank you. I suspected that I was wrong, but could not find anywhere where this is described.
